crt_portal/cts_forms/filters.py

```diff
@@ -61,12 +61,16 @@ def report_filter(request):
             elif filter_options[field] == '__contains':
                 kwargs[f'{field}__icontains'] = request.GET.getlist(field)[0]
             elif 'date' in field:
-                # filters by a start date or an end date expects YYYYMMDD
+                # filters by a start date or an end date expects yyyy-mm-dd
                 field_name = _get_date_field_from_param(field)
-                year = int(request.GET.getlist(field)[0][:4])
-                month = int(request.GET.getlist(field)[0][4:6])
-                day = int(request.GET.getlist(field)[0][6:])
-                kwargs[f'{field_name}{filter_options[field]}'] = datetime.date(year, month, day)
+                encodedDate = request.GET.getlist(field)[0]
+                decodedDate = urllib.parse.unquote(encodedDate)
+                dateObj = datetime.strptime(decodedDate, "%Y-%m-%d")
+                # if the year is before 2020, set it to 2020 because there is no data prior to 2020
+                year2020 = datetime.strptime("2020", "%Y")
```

Comment:
this was done so users don't pass non-parseable years that could cause failure in sql. Plus, there is no create date before 2020 in the system, since this is a new system, not a replacement system. users won't be confused, the date is displayed in the filter.

Reply:
Which failures in SQL are we talking about? I still think we can keep the year input as-is without loss of generality. 

Please note that while we may have no actual _production_ data before 2020, we certainly have data in staging and test environments before 2020 (and when doing load testing, we generate randomized dates). In these cases, preventing any date filter query before 2020 may be surprising to the user/developer.

Reply:
I've changed date input to date field with calendar widget, thus minimizing errors. this widget corrects most common typos. Only ill formatted dates are converted to 01/01/2020. The flip side is the display is in mm/dd/yyyy, that means the active filter display is in different format than date field. If we can convey this to Bill and Randy, we should be okay.